MediaGalleryMetadata/etc/di.xml

```diff
@@ -71,54 +71,54 @@
-            <argument name="fileWriter" xsi:type="object">Magento\MediaGalleryMetadata\Model\Jpeg\FileWriter</argument>
-            <argument name="writers" xsi:type="array">
-                <item name="xmp" xsi:type="object">Magento\MediaGalleryMetadata\Model\Jpeg\Segment\XmpWriter</item>
-                <item name="iptc" xsi:type="object">Magento\MediaGalleryMetadata\Model\Jpeg\Segment\IptcWriter</item>
+            <argument name="fileReader" xsi:type="object">Magento\MediaGalleryMetadata\Model\Jpeg\ReadFile</argument>
+            <argument name="fileWriter" xsi:type="object">Magento\MediaGalleryMetadata\Model\Jpeg\WriteFile</argument>
+            <argument name="metadataWriters" xsi:type="array">
+                <item name="xmp" xsi:type="object">Magento\MediaGalleryMetadata\Model\Jpeg\Segment\WriteXmp</item>
+                <item name="iptc" xsi:type="object">Magento\MediaGalleryMetadata\Model\Jpeg\Segment\WriteIptc</item>
             </argument>
         </arguments>
     </virtualType>
     <virtualType name="Magento\MediaGalleryMetadata\Model\Png\AddMetadata" type="Magento\MediaGalleryMetadata\Model\File\AddMetadata">
         <arguments>
-            <argument name="fileReader" xsi:type="object">Magento\MediaGalleryMetadata\Model\Png\FileReader</argument>
-            <argument name="fileWriter" xsi:type="object">Magento\MediaGalleryMetadata\Model\Png\FileWriter</argument>
-            <argument name="writers" xsi:type="array">
-                <item name="xmp" xsi:type="object">Magento\MediaGalleryMetadata\Model\Png\Segment\XmpWriter</item>
+            <argument name="fileReader" xsi:type="object">Magento\MediaGalleryMetadata\Model\Png\ReadFile</argument>
+            <argument name="fileWriter" xsi:type="object">Magento\MediaGalleryMetadata\Model\Png\WriteFile</argument>
+            <argument name="metadataWriters" xsi:type="array">
+                <item name="xmp" xsi:type="object">Magento\MediaGalleryMetadata\Model\Png\Segment\WriteXmp</item>
             </argument>
         </arguments>
     </virtualType>
     <virtualType name="Magento\MediaGalleryMetadata\Model\Gif\AddMetadata" type="Magento\MediaGalleryMetadata\Model\File\AddMetadata">
         <arguments>
-            <argument name="fileReader" xsi:type="object">Magento\MediaGalleryMetadata\Model\Gif\FileReader</argument>
-            <argument name="fileWriter" xsi:type="object">Magento\MediaGalleryMetadata\Model\Gif\FileWriter</argument>
-            <argument name="writers" xsi:type="array">
-                <item name="xmp" xsi:type="object">Magento\MediaGalleryMetadata\Model\Gif\Segment\XmpWriter</item>
+            <argument name="fileReader" xsi:type="object">Magento\MediaGalleryMetadata\Model\Gif\ReadFile</argument>
+            <argument name="fileWriter" xsi:type="object">Magento\MediaGalleryMetadata\Model\Gif\WriteFile</argument>
+            <argument name="metadataWriters" xsi:type="array">
```

Comment:
I would use more precise variable name (everything inside the module is related to metadata)
```suggestion
            <argument name="segmentWriters" xsi:type="array">
```